### ISSUE TITLE
Orphan fields (rows in database) found

### ISSUE THREAD

**Issue by tbba** [processwire/processwire-issues#282]
Question:
Is there a robust way with the API to identify field rows without corresponding entries in the "fields" database row and the backend fields list? (I could do this manually, of course, but there would be a risk of breaking things.)

Description of the problem:
1. I found field rows in some database (like "field_xyz") which have no corresponding name in the database's "fields" nor in the PW backend fields list (like field "xyz").

2. Also I found field rows with a "tmp_" prefix (like "tmp_field_xyz"), which also might be obsolete.

Those are fields I know I had deleted from fields using the PW backend. But obviously that has left  orphans behind in some cases. I was not able yet to reproduce when such a behavior occurs and maybe this is a leftover from older PW versions and fixed already.

That remaining field rows do not hurt. But they occupy space. And recreating a field with such a name would be impossible ("exists/violation" error).

PW 3.0.63, MySQL 6.5/ PHP 7

**Comment by ryancramerdesign** [comment]
The only time I've observed this is if I install ProcessWire on top of an existing database that already has tables in it. Are you sure that's not the case here? When installing PW, you ideally want to install into a new/blank database. As for the "tmp_" tables you mentioned, I don't know what those could be and haven't seen them before. But if they were "tmp_field_xyz" perhaps there was a failed field rename at some point? If something like that fails for some reason, it wouldn't be good for PW to go and delete tables, but should be fine for the end-user to do so if they've determined the data is not needed.

**Comment by tbba** [comment]
yes; I am sure the tmp_ files came from namings, that were rejected for a reason. Makes sense.
I think it was safe for me to delete them in the database.

Last time, I tried the profile export it could not handle multi language fields correctly.
While this was the case, I just duplicated a running project (for the 3 multilingual projects in question), deleted not needed content, templates fields and go from there. Is this wrong thinking? 

I was 100% sure to have deleted those fields from the backend, like normal. And I am now wondering why those field were gone in the backend and the database "fields" row (expected). 
But the individual field rows were not deleted somehow (not expected).

**Comment by ryancramerdesign** [comment]
> Last time, I tried the profile export it could not handle multi language fields correctly.

It should be able to handle them just fine. Or was it a non-core multi-language field of some sort?

> I just duplicated a running project (for the 3 multilingual projects in question), deleted not needed content, templates fields and go from there. Is this wrong thinking?

This is fine, as long as you are deleting from within PW and not directly in the database. 

> I was 100% sure to have deleted those fields from the backend, like normal. And I am now wondering why those field were gone in the backend and the database "fields" row (expected).
But the individual field rows were not deleted somehow (not expected).

I cant duplicate this here: creating, adding-to-template, populating, removing from template, then deleting field completely. The field_* table always gets deleted with the field. But if an error occurred when you were deleting a particular field, I can see how it might be possible. While I've not seen errors occur from deleting a field, it's always possible there is some site module hooking into something and causing an issue. However, the most common way for an orphan table to appear is if it was present already when PW was installed. If you find there are steps to consistently reproduce the issue, please let me know. Otherwise, I think it's safe for you to just manually delete the "tmp_*" or "field_*" tables you know aren't needed.

**Comment by tbba** [comment]
Looks like I found the reason for this. I checked my workflow and all tool Is use.
I am pretty sure now is outside of Processwire. 

In detail: It is the way the MySQL software NAVICAT handles data, when transferring databases (from DEV to LIVE, for example). It is not a full replace from one database with another but a replacement field by field. If I clean up fields in DEV and transfer the files to LIVE, the fields rows in use are updated but not touched fields in the target database left alone. Transferring files later opposite direction (backup for exampe) populates the orphan fields back to DEV. I will talk to the developer of NAVICAT about this. Solution could be an (optional) whiping of the target database bevor transferring data there.
